repair/repair.cc

```diff
@@ -2292,11 +2292,10 @@ future<> repair_service::repair_tablets(repair_uniq_id rid, sstring keyspace_nam
         while (true) {
             _repair_module->check_in_shutdown();
             erm = t->get_effective_replication_map();
-            const locator::tablet_map& tmap = erm->get_token_metadata_ptr()->tablets().get_tablet_map(tid);
-            if (!tmap.has_transitions()) {
+            if (co_await container().invoke_on(0, [] (repair_service& rs) { return !rs._tsm.local()._topology.is_busy(); })) {
```

Comment:
There is a loss of atomicity here, the version of erm we obtained may not be the same as the version of topology on which we check business. So the transitions may have ended on shard 0 but this erm still contains tablets in transition. This can give bad results if the code later is assuming no transitions and looks at "replicas" - the tablet may already migrated to a different replica set, but it's not reflected in erm yet (only in transition state).